File: go-awesome/go-proj1/src/util/webp2b64str.py
import base64


def webp2b64Str(inFilePath):
  with open(inFilePath, "rb") as webp:
    b64EncStr = base64.b64encode(webp.read())
  return b64EncStr.decode('utf-8')


def write2txt(b64EncStr, outFilePath):
  with open(outFilePath, "w") as txt:
    txt.write(b64EncStr)


if __name__ == '__main__':
  write2txt(webp2b64Str('./qiqi.webp'), './qiqi.txt')

File: go-awesome/go-proj1/src/util/webp2b64str.go
package main

import (
	"encoding/base64"
	"os"
)

func webp2b64Str(inFilePath string) string {
	data, err := os.ReadFile(inFilePath)
	if err != nil {
		panic(err)
	}
	b64EncStr := base64.StdEncoding.EncodeToString(data)
	return b64EncStr
}

func write2txt(b64EncStr string, outFilePath string) {
	err := os.WriteFile(outFilePath, []byte(b64EncStr), 0644)
	if err != nil {
		panic(err)
	}
}

func main() {
	b64EncStr := webp2b64Str("./qiqi.webp")
	write2txt(b64EncStr, "../dao/qiqi.txt")
}
